From 539f01ad408471cd4493a31cba92872a08eebbb7 Mon Sep 17 00:00:00 2001
From: chaviw <chaviw@google.com>
Date: Wed, 10 Jan 2018 12:31:08 -0800
Subject: [PATCH 052/306] Revert "Revert "Revert "Avoid setting size/position
 on Transaction if not needed"""

This reverts commit 1e8c4c8c0c9111b7ae90e110c73122c7bff1df81.

Fixes: 71789344
(cherry picked from commit b38a6704e09b80b9b4fe4f3ad0d16ad6179f612e)

Change-Id: Ibfa7408d990e07fa5b299e7edd34c02cfe44be5b
---
 .../core/java/com/android/server/wm/TaskStack.java     | 10 +---------
 .../java/com/android/server/wm/WindowContainer.java    |  6 ------
 .../core/java/com/android/server/wm/WindowState.java   |  4 +---
 3 files changed, 2 insertions(+), 18 deletions(-)

diff --git a/services/core/java/com/android/server/wm/TaskStack.java b/services/core/java/com/android/server/wm/TaskStack.java
index 7b4281c61f1..bdda944f236 100644
--- a/services/core/java/com/android/server/wm/TaskStack.java
+++ b/services/core/java/com/android/server/wm/TaskStack.java
@@ -44,7 +44,6 @@ import static com.android.server.wm.proto.StackProto.WINDOW_CONTAINER;
 
 import android.annotation.CallSuper;
 import android.content.res.Configuration;
-import android.graphics.Point;
 import android.graphics.Rect;
 import android.graphics.Region;
 import android.os.RemoteException;
@@ -146,7 +145,6 @@ public class TaskStack extends WindowContainer<Task> implements
      * For {@link #prepareSurfaces}.
      */
     final Rect mTmpDimBoundsRect = new Rect();
-    private final Point mLastSurfaceSize = new Point();
 
     TaskStack(WindowManagerService service, int stackId, StackWindowController controller) {
         super(service);
@@ -746,13 +744,7 @@ public class TaskStack extends WindowContainer<Task> implements
         }
 
         final Rect stackBounds = getBounds();
-        final int width = stackBounds.width();
-        final int height = stackBounds.height();
-        if (width == mLastSurfaceSize.x && height == mLastSurfaceSize.y) {
-            return;
-        }
-        transaction.setSize(mSurfaceControl, width, height);
-        mLastSurfaceSize.set(width, height);
+        transaction.setSize(mSurfaceControl, stackBounds.width(), stackBounds.height());
     }
 
     @Override
diff --git a/services/core/java/com/android/server/wm/WindowContainer.java b/services/core/java/com/android/server/wm/WindowContainer.java
index 36e6418a39b..5548f3d7278 100644
--- a/services/core/java/com/android/server/wm/WindowContainer.java
+++ b/services/core/java/com/android/server/wm/WindowContainer.java
@@ -95,7 +95,6 @@ class WindowContainer<E extends WindowContainer> extends ConfigurationContainer<
     protected final WindowManagerService mService;
 
     private final Point mTmpPos = new Point();
-    protected final Point mLastSurfacePosition = new Point();
 
     /** Total number of elements in this subtree, including our own hierarchy element. */
     private int mTreeWeight = 1;
@@ -1179,12 +1178,7 @@ class WindowContainer<E extends WindowContainer> extends ConfigurationContainer<
         }
 
         getRelativePosition(mTmpPos);
-        if (mTmpPos.equals(mLastSurfacePosition)) {
-            return;
-        }
-
         transaction.setPosition(mSurfaceControl, mTmpPos.x, mTmpPos.y);
-        mLastSurfacePosition.set(mTmpPos.x, mTmpPos.y);
 
         for (int i = mChildren.size() - 1; i >= 0; i--) {
             mChildren.get(i).updateSurfacePosition(transaction);
diff --git a/services/core/java/com/android/server/wm/WindowState.java b/services/core/java/com/android/server/wm/WindowState.java
index 0ad60c93bb4..b3809dd8f6c 100644
--- a/services/core/java/com/android/server/wm/WindowState.java
+++ b/services/core/java/com/android/server/wm/WindowState.java
@@ -4487,7 +4487,6 @@ class WindowState extends WindowContainer<WindowState> implements WindowManagerP
 
         // Leash is now responsible for position, so set our position to 0.
         t.setPosition(mSurfaceControl, 0, 0);
-        mLastSurfacePosition.set(0, 0);
     }
 
     @Override
@@ -4503,9 +4502,8 @@ class WindowState extends WindowContainer<WindowState> implements WindowManagerP
         }
 
         transformFrameToSurfacePosition(mFrame.left, mFrame.top, mSurfacePosition);
-        if (!mSurfaceAnimator.hasLeash() && !mLastSurfacePosition.equals(mSurfacePosition)) {
+        if (!mSurfaceAnimator.hasLeash()) {
             t.setPosition(mSurfaceControl, mSurfacePosition.x, mSurfacePosition.y);
-            mLastSurfacePosition.set(mSurfacePosition.x, mSurfacePosition.y);
         }
     }
 
-- 
2.17.1

